src/domain/session/room/timeline/tiles/BaseTextTile.js

```diff
@@ -56,4 +57,43 @@ export class BaseTextTile extends BaseMessageTile {
         }
         return this._messageBody;
     }
+
+    get replyPreviewBody() {
+        if (!this._entry.contextEventId) {
+            return null;
+        }
+        const entry = this._entry.contextEntry;
+        const error = this._generateError(entry);
+        if (error?.name === "ContextEntryNotFound") {
+            return { error };
+        }
+        const format = entry.content.format === "org.matrix.custom.html" ? BodyFormat.Html : BodyFormat.Plain;
+        const body = entry.content["formatted_body"] ?? entry.content["body"];
+        return {
+            body: body && this._parseBody(body, format),
+            senderName: entry.displayName,
+            avatar: {
+                avatarColorNumber: getIdentifierColorNumber(entry.sender),
+                avatarLetter: avatarInitials(entry.displayName),
+                avatarUrl: (size) => getAvatarHttpUrl(entry.avatarUrl, size, this.platform, this._mediaRepository)
+            },
+            error
+        };
+    }
+
+    _generateError(entry) {
```

Comment:
I'd perhaps be more inclined to work with flags on the proposed ReplyPreviewViewModel class rather than using Error objects for this.

Reply:
The not found case we can handle if the reply entry has `isReply === true` but there is no contextEvent.